common/src/main/scala/explore/utils/package.scala

```diff
@@ -170,3 +176,66 @@ extension (bytes: NonNegLong)
 
 object IsExpanded extends NewType[Boolean]
 type IsExpanded = IsExpanded.Type
+
+extension [F[_], O](s: Stream[F, O])
+  /**
+   * Combine items within a given duration using the given function.
+   *
+   * When an item is received in the stream, a timer is started. Within that time, items are
+   * combined until the timer expires, at which point the combined item is emitted.
+   *
+   * If no items are received within the given duration, no item is emitted.
+   */
+  def reduceWithin(d: FiniteDuration, f: (O, O) => O)(using F: Temporal[F]): Stream[F, O] =
+    given Semigroup[O] with
+      def combine(x: O, y: O): O = f(x, y)
+    reduceSemigroupWithin(d)
```

Comment:
Nit, but I don't think it makes sense to express the implementations in this direction. Not every "reducer" `f` is a lawful `Semigroup`.

Reply:
Hmm, okay. It's mostly this way because I wanted the syntax of `combineAllOption`, which I would then have to write out manually (unless you have a suggestion for that)